src/axom/inlet/Field.cpp

```diff
@@ -56,14 +58,189 @@ bool Field::required()
     SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
     return false;
   }
 
   return (bool)intValue;
 }
 
+template <typename T>
+void Field::setDefaultValue(T value) {
+  if (m_sidreGroup->hasView("defaultValue")) {
+    std::string msg = fmt::format("Inlet Field has already defined default value: {0}",
+                                  m_sidreGroup->getPathName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    if (m_docEnabled) {
+      m_sidreGroup->createViewScalar("defaultValue", value);
+    }
+    if (!m_sidreGroup->hasView("value")) {
+      m_sidreGroup->createViewScalar("value", value);
+    }
+  }
+}
+
+template<> 
+void Field::setDefaultValue<std::string>(std::string value) {
+  if (m_sidreGroup->hasView("defaultValue")) {
+    std::string msg = fmt::format("Inlet Field has already defined default value: {0}",
+                                  m_sidreGroup->getPathName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    if (m_docEnabled) {
+      m_sidreGroup->createViewString("defaultValue", value);
+    }
+    if (!m_sidreGroup->hasView("value")) {
+      m_sidreGroup->createViewString("value", value);
+    }
+  }
+}
+
+std::shared_ptr<Field> Field::defaultValue(const char* value) {
+  std::string str = "";
+  if (value) {
+    str = value;
+  }
+  setDefaultValue(std::string(value));
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(const std::string& value) {
+  if (m_type != axom::sidre::DataTypeId::CHAR8_STR_ID) {
+    SLIC_WARNING("Field value type did not match STRING");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  setDefaultValue(value);
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(bool value) {
+  if (m_type != axom::sidre::DataTypeId::INT8_ID) {
+    SLIC_WARNING("Field value type did not match BOOL");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  setDefaultValue((int8)value);
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(int value) {
+  switch (m_type) {
+  case axom::sidre::DataTypeId::INT_ID:
+    setDefaultValue(value);
+    break;
+  case axom::sidre::DataTypeId::DOUBLE_ID:
+    setDefaultValue((double)value);
+    break;
+  default:
+    SLIC_WARNING("Field value type did not match INT/DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+    break;
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(double value) {
+  if (m_type != axom::sidre::DataTypeId::DOUBLE_ID) {
+    SLIC_WARNING("Field value type did not match DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  setDefaultValue(value);
+  return shared_from_this();
+}
+
+template<typename T>
+void Field::setRange(T startVal, T endVal) {
+  if (m_sidreGroup->hasView("range") || m_sidreGroup->hasView("validValues")
+      || m_sidreGroup->hasView("validStringValues")) {
+    std::string msg = fmt::format("Inlet Field has already defined range: {0}",
```

Comment:
This warning message should be broken into three separate if's to clarify to the user what they need to fix:

```
if (m_sidreGroup->hasView("range")) {
    std::string msg = fmt::format("Inlet Field has already defined range: {0}", m_sidreGroup->getPathName());
    SLIC_WARNING(msg);
    setWarningFlag(m_sidreRootGroup);
}
else if(m_sidreGroup->hasView("validValues") || m_sidreGroup->hasView("validStringValues")){
    std::string msg = fmt::format("Cannot set range on Inlet Field after setting valid value: {0}", m_sidreGroup->getPathName());
    SLIC_WARNING(msg);
    setWarningFlag(m_sidreRootGroup);
}
```